src/Generation/faction.py

```diff
@@ -32,34 +46,48 @@ def _create_faction(scenario: Scenario, power: int, stance: InitialStance, align
     results = llm_create('faction', count,
                capital_name=scenario.capital.name,
                kingdom=scenario.kingdom.name,
-               tyrant=scenario.ruler.name,
+               ruler=scenario.ruler.name,
                ruler_personality=scenario.ruler.personality,
                ruler_evil=scenario.ruler.evil_deeds,
                governance_style=scenario.ruler.governance_style,
                background_kingdom=scenario.kingdom.description,
                background_capital=f'{scenario.capital.history} {scenario.capital.architecture} {scenario.capital.population}',
                power_level=power_level,
                rebellion_stance=stance.value,
-               alignment=alignment
+               alignment=alignment.value
                )
-    return [Faction(name=result.name, overview=result.overview, motto=result.motto, beliefs=result.beliefs, goals=result.goals,
-                    needs=result.needs) for result in results]
+    factions = []
+    for result in results:
+        npcs = []
+        faction = Faction(name=result.name, overview=result.overview, motto=result.motto,
+                                beliefs=result.beliefs, goals=result.goals, needs=result.needs,
+                                power=power_level, power_value=power, npcs=npcs)
+        factions.append(faction)
+        npcs.extend(create_npcs(scenario, faction))
+    return factions
 
 
 def create_factions(scenario: Scenario) -> list[Faction]:
     factions = []
 
     # Create a few friendly, but not very powerful factions
-    factions.append(_create_faction(scenario, 5, InitialStance.FRIENDLY, 'chaotic good'))
-    factions.append(_create_faction(scenario, 8, InitialStance.FRIENDLY, 'neutral good'))
-    factions.append(_create_faction(scenario, 20, InitialStance.FRIENDLY, 'chaotic evil'))
+    factions.extend(_create_faction(scenario, 5, InitialStance.FRIENDLY, Alignment.CHAOTIC_GOOD, 1))
+    # factions.extend(_create_faction(scenario, 15, InitialStance.FRIENDLY, Alignment.NEUTRAL_GOOD, 2))
```

Comment:
Are these commented out here for "financial reasons"?

Reply:
Actually, this is mostly to improve performance - generating a ton of text with GPT-3.5 is cheap, but not very _fast_.